package.json

```diff
@@ -119,7 +119,8 @@
   },
   "scripts": {
     "postinstall": "yarn theme",
-    "build": "gatsby build",
+    "build": "yarn type-check && yarn build:gatsby",
```

Comment:
I was thinking about your comment earlier about the issue with this where it would break running the first time until someone has built the project. What if we used a conditional in that build command

```
"build": "if [ -f path/to/file.js ]; then yarn type-check && yarn build:gatsby; else yarn build:gatsby && yarn type-check; fi"
```

This way, we could check for the `gatsby-types.d.ts` file, if it exists run type-check first and then gatsby:build. If the types file isn't there, then we would build first then check types. 

I think this may address your concerns that we were talking about earlier.

Reply:
Is this something we're still looking to implement @pettinarip?

Reply:
@wackerow @corwintines I'd say that given that we are in the transition to move to NextJS we can discard this change for now and enforce this again when the migration is done.

Will close this PR. Let me know if you disagree.